mcs/build/rules.make

```diff
@@ -136,11 +136,13 @@ endif
 # mcs/class/lib/$(PROFILE)/
 ifndef TOP_LEVEL_DO
 
-ifdef ALWAYS_AOT
+ifdef ALWAYS_AOT_BCL
```

Comment:
I don't think it's working as you would expect it to, you don't have _AFAIK_ a `else if` in Makefile, and you can only nest the conditions inside `else` blocks.

If you want to avoid repetition, you should need the following:
```
ifneq ($(or $(ALWAYS_AOT_BCL),$(ALWAYS_AOT_TESTS)),)
TOP_LEVEL_DO = do-all-aot
else
TOP_LEVEL_DO = do-all
endif
```